fabtests/common/shared.c

```diff
@@ -782,7 +782,21 @@ int ft_init(void)
 		return ret;
 	}
 
-	return ft_hmem_init(opts.iface);
+	ret = ft_hmem_init(opts.iface);
+	if (ret != FI_SUCCESS) {
+		FT_ERR("ft_hmem_init: %d", ret);
+		return ret;
+	}
+
+	if (opts.iface == FI_HMEM_CUDA) {
```

Comment:
shouldn't we do this in `fabtests/common/hmem_cuda.c`?

Reply:
neuron does this in `ft_neuron_alloc`, we should do it similarly in `ft_cuda_alloc`.

Reply:
I think it makes sense in ft_cuda_init().  Likely don't need to expose ft_cuda_set_device().

Reply:
I moved it to inside `ft_cuda_init()` in new revision, also removed `ft_cuda_set_device()`